Content.Server/Chat/TelepathicChatSystem.Psychognomy.cs

```diff
@@ -0,0 +1,187 @@
+            SubscribeLocalEvent<MetempsychosisKarmaComponent, GetPsychognomicDescriptorEvent>(DescribeKarma);
+            SubscribeLocalEvent<GlimmerSourceComponent, GetPsychognomicDescriptorEvent>(DescribeGlimmerSource);
+            SubscribeLocalEvent<PsionicComponent, GetPsychognomicDescriptorEvent>(DescribePsion);
+            SubscribeLocalEvent<InnatePsionicPowersComponent, GetPsychognomicDescriptorEvent>(DescribeInnatePsionics);
+            SubscribeLocalEvent<BloodSuckerComponent, GetPsychognomicDescriptorEvent>(DescribeBloodsucker);
+            SubscribeLocalEvent<RevenantComponent, GetPsychognomicDescriptorEvent>(DescribeRevenant);
+            // SubscribeLocalEvent<GlimmerWispComponent, GetPsychognomicDescriptorEvent>(DescribeGlimmerWisp);
+            SubscribeLocalEvent<FamiliarComponent, GetPsychognomicDescriptorEvent>(DescribeFamiliar);
+            // SubscribeLocalEvent<GolemComponent, GetPsychognomicDescriptorEvent>(DescribeGolem);
+            SubscribeLocalEvent<OracleComponent, GetPsychognomicDescriptorEvent>(DescribeOracle);
+            SubscribeLocalEvent<SophicScribeComponent, GetPsychognomicDescriptorEvent>(DescribeSophia);
+        }
+
+        private void DescribeHumanoid(EntityUid uid, HumanoidAppearanceComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            if (component.Sex != Sex.Unsexed)
+                ev.Descriptors.Add(component.Sex == Sex.Male ? Loc.GetString("p-descriptor-male") : Loc.GetString("p-descriptor-female"));
+
+            // Deliberately obfuscating a bit; psionic signatures use different standards
+            ev.Descriptors.Add(component.Age >= 100 ? Loc.GetString("p-descriptor-old") : Loc.GetString("p-descriptor-young"));
+        }
+
+        private void DescribeGrammar(EntityUid uid, GrammarComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            if (component.Gender == Gender.Male || component.Gender == Gender.Female)
+                ev.Descriptors.Add(component.Gender == Gender.Male ? Loc.GetString("p-descriptor-masculine") : Loc.GetString("p-descriptor-feminine"));
+        }
+
+        private void DescribeDamage(EntityUid uid, DamageableComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            if (component.DamageContainerID == "CorporealSpirit")
+            {
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-liminal"));
+                if (!HasComp<HumanoidAppearanceComponent>(uid))
+                    ev.Descriptors.Add(Loc.GetString("p-descriptor-old"));
+                return;
+            }
+
+            ev.Descriptors.Add(Loc.GetString("p-descriptor-hylic"));
+        }
+
+        private void DescribeMobState(EntityUid uid, MobStateComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            if (component.CurrentState == Shared.Mobs.MobState.Critical)
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-liminal"));
+        }
```

Comment:
```suggestion
        private void DescribeMobState(EntityUid uid, MobStateComponent component, GetPsychognomicDescriptorEvent ev)
        {
            if (component.CurrentState != Shared.Mobs.MobState.Critical)
                return;

            ev.Descriptors.Add(Loc.GetString("p-descriptor-liminal"));
        }
```
These are technically going to increase the line count slightly, but are still this repo's standard coding practice to invert logic to get rid of indents. It's also technically more efficient. :)